webapp/src/components/systemConsole/teamFilter/teamFilter.tsx

```diff
@@ -32,55 +32,84 @@ const TEAM_FILTER_FILTER_TYPE_OPTIONS: RadioSetting[] = [
 
 function TeamFilter({id, setSaveNeeded, onChange, config, setInitialSetting}: CustomSettingChildComponentProp) {
     const [selectedTeams, setSelectedTeams] = useState<DropdownOption[]>([]);
-    const [allTeamsOptions, setAllTeamsOptions] = useState<DropdownOption[]>([]);
+    const [initialOptions, setInitialOptions] = useState<DropdownOption[]>([]);
     const [teamFilterType, setTeamFilterType] = useState<TeamFilterType>('everyone');
 
+    // For fetching initial few teams for displaying in the multiselect dropdown
     useEffect(() => {
         const task = async () => {
-            // fetch all teams to populate options
-            const teams: Team[] = await Client4.getTeams(0, 10000, false) as Team[];
+            // getTeams always returns Team[] when includeTotalCount param is set to false
+            const teams = await Client4.getTeams(0, 20, false) as Team[];
```

Comment:
is this standard way of doing this in this repo. In web we usually have an action.